stdlib/SparseArrays/src/higherorderfns.jl

```diff
@@ -810,6 +815,49 @@ end
 _finishempty!(C::SparseVector) = C
 _finishempty!(C::SparseMatrixCSC) = (fill!(C.colptr, 1); C)
 
+# special case - vector outer product
+_copy(f::typeof(*), x::SparseVectorUnion, y::AdjOrTransSparseVectorUnion) = _outer(x, y)
+@inline _outer(x::SparseVectorUnion, y::Adjoint) = return _outer(conj, x, y)
+@inline _outer(x::SparseVectorUnion, y::Transpose) = return _outer(identity, x, y)
+function _outer(trans::Tf, x, y) where Tf
+    w = parent(y)
```

Comment:
I'd find this a bit more readable if the `parent` unwrapping happened in the above two methods that add the function argument.  Not a big deal, but I was confused for a moment thinking you'd `conj` `y` twice.  This way, your core function is simply the outer product between two sparse vectors.

Reply:
Yeah, good suggestion. I'd originally had a `y isa Adjoint` check in the loop, but captured on the idea of using multiple dispatch to make the decision instead — just didn't fully commit to the idea. :-) Changed.